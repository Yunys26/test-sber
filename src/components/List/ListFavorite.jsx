import React from 'react';
// Libs
import {
    Typography,
    Grid,
    IconButton,
    Box, 
    CssBaseline, 
    ThemeProvider
} from '@material-ui/core';
import FavoriteIcon from '@material-ui/icons/Favorite';
import PropTypes from 'prop-types';
import moment from 'moment';
import { useSelector } from 'react-redux';
// Theme
import { themeSearchBlock } from '../MainBlock/mainBlockTheme';
import { themeTabs } from '../TabsBlock/tabsBlockTheme';

export default function ListFavorite(props) {

    const { classes, funcLogic, statusListFavorit } = props;

    const local = useSelector(state => state.main.localStorageStore);

    return (
        <div>
            {
                local.length === 0 && statusListFavorit || Object.values(...local).map((el) => 
                    <Box className={classes.blockContent}>
                        <Box className={classes.workBlock} boxShadow={4} id={el.id} >
                            
                            <Grid 
                                container 
                                item={true} 
                                xs={12}
                                className={classes.titleSearchBlock} 
                                direction="row" 
                                justify="space-between" 
                                alignItems="center"
                                >
                                    <Typography className={classes.nameCompany} variant="h3">{el.company}</Typography>
                                    {(el.company_logo !== null && <img className={classes.workBlockLogo} src={el.company_logo} alt=""></img>) || null}
                                    <ThemeProvider theme={themeSearchBlock}>
                                    <CssBaseline/>
                                        <IconButton
                                            style={{color: "#f50057"}} 
                                            className={classes.likeBlack} 
                                        >
                                            <FavoriteIcon className={classes.likeBlack} />
                                        </IconButton>
                                    </ThemeProvider>
                            </Grid>
                            
                            <Typography variant="h5"><b>Title:</b> {el.title}</Typography>
                            <Typography variant="h5"><b>Company:</b> {el.company}</Typography>
                            <Typography variant="h5"><b>Location:</b> {el.location}</Typography>
                            <Typography variant="h5"><b>Type:</b> {el.type}</Typography>
                            <Typography variant="h5"><b>Location:</b> {el.location}</Typography>
                            <ThemeProvider theme={themeTabs}>
                            <CssBaseline/>
                                <div className={classes.descriptionBlock} dangerouslySetInnerHTML={{ __html: el.description }}></div>
                                <div className={classes.descriptionBlock} dangerouslySetInnerHTML={{ __html: el.how_to_apply }}></div>
                            </ThemeProvider>
                            <Typography className={classes.footerSearchBlock} variant="h6">{local.length !== 1 && moment(el.created_at).format('LLLL')}</Typography>
                        
                        </Box>
                        
                    </Box>
                )
            }
        </div>
    )
}

ListFavorite.propTypes = {
    classes: PropTypes.object,
    funcLogic: PropTypes.func,
    statusListFavorit: PropTypes.object,
}