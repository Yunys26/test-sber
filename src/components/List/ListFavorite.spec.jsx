import React, { useEffect } from 'react';
import { mount, shallow } from 'enzyme';
import ListFavorite from './ListFavorite';
import { Button } from '@material-ui/core';

// Мокаем useDispatch
jest.mock('react-redux', () => ({
    useDispatch: () => jest.fn()
}));


describe('<ListFavorite />', () => {
    let props, componentListFavorite;

    beforeEach(() => {

        props = {
            classes: {},
            statusList: {},
            local: [
                {
                    company: "NS",
                    company_logo: "https://jobs.github.com/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBdUdMIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--9413d6855d50ab98787e3589f63491eacd6c7e0b/NS%20CB.jpg",
                    company_url: "http://www.ns.nl",
                    created_at: "Tue Oct 13 10:25:20 UTC 2020",
                    description: "senior iOS developer NS App - Utrecht Utrecht",
                    favorite: false,
                    how_to_apply: "Solliciteer direct via de",
                    id: "65197190-0116-414a-9406-ae150edccb57",
                    location: "Utrecht",
                    title: "senior iOS developer NS App ",
                    type: "Full Time",
                    url: "https://jobs.github.com/positions/65197190-0116-414a-9406-ae150edccb57",
                },
            ],
        };

        componentListFavorite = shallow(<ListFavorite {...props} />)
    });

    afterEach(() => {
        props, componentListFavorite = undefined;
    });

    it('Проверка на рендер <ListFavorite />', () => {
        expect(componentListFavorite.length).toBe(1);
    });

    it('Snapshot ListFavorite', () => {
        expect(componentListFavorite).toMatchSnapshot();
    });
}) 