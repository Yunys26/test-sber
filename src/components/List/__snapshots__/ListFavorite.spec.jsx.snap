// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListFavorite /> Snapshot ListFavorite 1`] = `undefined`;

exports[`<ListFavorite /> Snapshot ListFavorite 2`] = `undefined`;
